src/Middleware/ValidateWebAppData.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Nutgram\Laravel\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use SergiX44\Nutgram\Exception\InvalidDataException;
+use SergiX44\Nutgram\Nutgram;
+
+class ValidateWebAppData
+{
+    public function handle(Request $request, Closure $next)
+    {
+        try {
+            $initData = $request->input('initData', '');
+            $data = app(Nutgram::class)->validateWebAppData($initData);
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common processes. Try to leave a link to the process described so that the reader knows how to resolve the chore.  -->
**💣 chore:** ‏ Instead of manual resolving the instance I would use the DI on the constructor